src/db.rs

```diff
@@ -216,6 +216,42 @@ pub async fn state_update(
     Ok(data_product)
 }
 
+/// Retrieve a Data Product
+pub async fn data_product_select(
```

Comment:
@coderabbitai Please add a unit test where this function can read a data product record from the DB, and a unit test where there data product does not exists.

Reply:
Please create a PR for this change.